src/instana/instrumentation/urllib3.py

```diff
@@ -2,109 +2,137 @@
-
-            if type(instance) is urllib3.connectionpool.HTTPSConnectionPool:
-                kvs['url'] = 'https://%s:%d%s' % (kvs['host'], kvs['port'], kvs['path'])
+                    kvs["query"] = strip_secrets_from_query(
+                        parts[1],
+                        agent.options.secrets_matcher,
+                        agent.options.secrets_list,
+                    )
+
+            if isinstance(instance, urllib3.connectionpool.HTTPSConnectionPool):
+                kvs["url"] = "https://%s:%d%s" % (kvs["host"], kvs["port"], kvs["path"])
             else:
-                kvs['url'] = 'http://%s:%d%s' % (kvs['host'], kvs['port'], kvs['path'])
+                kvs["url"] = "http://%s:%d%s" % (kvs["host"], kvs["port"], kvs["path"])
         except Exception:
-            logger.debug("urllib3 collect error", exc_info=True)
+            logger.debug("urllib3 _collect_kvs error", exc_info=True)
             return kvs
         else:
             return kvs
 
-
-    def collect_response(scope, response):
+    def collect_response(
+        span: "InstanaSpan", response: urllib3.response.HTTPResponse
+    ) -> None:
         try:
-            scope.span.set_tag(ext.HTTP_STATUS_CODE, response.status)
+            span.set_attribute(SpanAttributes.HTTP_STATUS_CODE, response.status)
 
-            extract_custom_headers(scope.span, response.headers)
+            _extract_custom_headers(span, response.headers)
 
             if 500 <= response.status:
-                scope.span.mark_as_errored()
+                span.mark_as_errored()
         except Exception:
             logger.debug("collect_response", exc_info=True)
 
+    @wrapt.patch_function_wrapper("urllib3", "HTTPConnectionPool.urlopen")
+    def urlopen_with_instana(
+        wrapped: Callable[
+            ..., Union[urllib3.HTTPConnectionPool, urllib3.HTTPSConnectionPool]
+        ],
+        instance: Union[
+            urllib3.connectionpool.HTTPConnectionPool,
+            urllib3.connectionpool.HTTPSConnectionPool,
+        ],
+        args: Tuple[int, str, Tuple[Any, ...]],
+        kwargs: Dict[str, Any],
+    ) -> urllib3.response.HTTPResponse:
+        tracer, parent_span, span_name = get_tracer_tuple()
 
-    @wrapt.patch_function_wrapper('urllib3', 'HTTPConnectionPool.urlopen')
-    def urlopen_with_instana(wrapped, instance, args, kwargs):
-        tracer, parent_span, operation_name = get_tracer_tuple()
         # If we're not tracing, just return; boto3 has it's own visibility
-        if (tracing_is_off() or (operation_name == 'boto3')):
+        if tracing_is_off() or (span_name == "boto3"):
             return wrapped(*args, **kwargs)
 
-        with tracer.start_active_span("urllib3", child_of=parent_span) as scope:
+        parent_context = parent_span.get_span_context() if parent_span else None
+
+        with tracer.start_as_current_span(
+            "urllib3", span_context=parent_context, kind=SpanKind.CLIENT
```

Comment:
As discussed yesterday, can we remove this?

```suggestion
            "urllib3", span_context=parent_context
```